cpp_linter/clang_tools/clang_tidy.py

```diff
@@ -260,10 +261,33 @@ def run_clang_tidy(
     advice = parse_tidy_output(results.stdout.decode(), database=db_json)
 
     if tidy_review:
-        # store the modified output from clang-tidy
-        advice.patched = Path(file_obj.name).read_bytes()
-        # re-write original file contents
-        Path(file_obj.name).write_bytes(original_buf)
+        timeout = time.monotonic_ns() + 1000000
+        # wait 1s for file to become accessible
+        success = False
+        while time.monotonic_ns() < timeout:
+            try:
+                src = open(filename, "r+b")
```

Comment:
I guess this timeout idea works reliably?

Would really be great to have a reproducer.

Reply:
It fixes the issue from the code base in the original issue ticket. I tried it several times with the header file changed to have tidy fixes found and with no tidy fixes to be found and all times the review output the correct information.
Not sure how easy it is to make a public reproducible test case, in theory anything with modified cpp files that all include the same header and that header has also been modified. Run it on a windows github action and see what it does. Im not familiar with your test suite so ill leave that to you